dali/operators/reader/loader/video/video_test.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/loader/video/video_test.h"
+
+#include <opencv2/imgproc.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <fstream>
+
+#include "dali/test/dali_test_config.h"
+#include "dali/test/cv_mat_utils.h"
+
+namespace dali {
+
+// Define static tests members - needed to hold resources between tests
+std::vector<std::vector<cv::Mat>> VideoTest::cfr_frames_;
+std::vector<std::vector<cv::Mat>> VideoTest::vfr_frames_;
+
+void VideoTest::SetUpTestSuite() {
+  std::vector<std::string> cfr_frames_paths{
+      testing::dali_extra_path() + "/db/video/cfr/frames_1/",
+      testing::dali_extra_path() + "/db/video/cfr/frames_2/"};
+
+  std::vector<std::string> vfr_frames_paths{
+      testing::dali_extra_path() + "/db/video/vfr/frames_1/",
+      testing::dali_extra_path() + "/db/video/vfr/frames_2/"};
+
+  
+  LoadFrames(cfr_frames_paths, cfr_frames_);
+  LoadFrames(vfr_frames_paths, vfr_frames_);
+}
+
+void VideoTest::LoadFrames(std::vector<std::string> &paths, std::vector<std::vector<cv::Mat>> &out_frames) {
+  for (auto &frames_path : paths) {
+    std::vector<cv::Mat> frames;
+    std::ifstream frames_list(frames_path + "frames_list.txt");
+    std::string frame_filename;
+
+    while (std::getline(frames_list, frame_filename)) {
+      cv::Mat frame;
+      cv::cvtColor(cv::imread(frames_path + frame_filename), frame, cv::COLOR_BGR2RGB);
+      frames.push_back(frame);
+    }
+
+    out_frames.push_back(frames);
+  }
+}
+
```

Comment:
Can you add a test with more than one video file opened in parallel and seeking to a random frame in the random video each time to mimic what we do now in DALI?

Reply:
This is tested at the level of the reader: https://github.com/awolant/DALI/blob/video_reader_cpu/dali/operators/reader/video_reader_cpu_op_test.cc#L31

If you think it is useful to test it here as well. It would be helpful, if the reader test failed and video file test didn't. It would indicate that there is something wrong with reader/loader rather then the video file. I'm just not sure is it worth to test it two times in every pipeline ever. After all, we assume that this will work more often than not.

Reply:
Sure.